rpcclient/wallet.go

```diff
@@ -826,6 +828,33 @@ func (c *Client) PurchaseTicket(fromAccount string,
 
 // END DECRED FUNCTIONS -----------------------------------------------------------
 
+// decodeNetParams houses the well-known network parameters to use when
+// attempting to decode addresses and WIF strings.
+var decodeNetParams = []*chaincfg.Params{
+	chaincfg.MainNetParams(),
```

Comment:
This goes against our previous discussion about how most programs should only call the params functions once in their lifetime.  I think passing the parameters to client methods is a better way.

This will also decode addresses for the wrong network and dcrutil.Address no longer has any way to return that network to the user or check that it is on a correct network.

Reply:
Agreed.  I wasn't really happy with that as I wrote it, but since it didn't have access to the params at that point, I decided to go with that approach.

I'll change it over to take a `dcrutil.AddressParams` parameter in the client func as well as the ` Receive` func on the relevant futures.

Reply:
Updated.